lib/bosh/BoshDirectorClient.js

```diff
@@ -358,33 +480,41 @@ class BoshDirectorClient extends HttpClient {
     const query = _.assign({
       limit: 1000
     }, options);
-    return this
-      .request({
-        method: 'GET',
-        url: '/tasks',
-        qs: _.pick(query, ['limit', 'state', 'deployment'])
-      }, 200)
-      .then(res => JSON.parse(res.body));
+
+    return Promise
+      .map(this.activeConfigs, directorConfig => this
+        .makeRequestWithConfig({
+          method: 'GET',
+          url: '/tasks',
+          qs: _.pick(query, ['limit', 'state', 'deployment'])
+        }, 200, directorConfig)
+        .then(res => JSON.parse(res.body))
+      )
+      .reduce((all_tasks, tasks) => all_tasks.concat(tasks), []);
   }
 
   getTask(taskId) {
+    var deploymentName = this.parseTaskid(taskId, 1);
+    var taskIdAlone = this.parseTaskid(taskId, 2);
```

Comment:
inefficient. `parseTaskId` must return an array and here in the invocation points just lookup index. Regex is matched everytime in `parseTaskid` method.